lib/travis/build/addons/browserstack.rb

```diff
@@ -0,0 +1,153 @@
+require 'travis/build/addons/base'
+
+module Travis
+  module Build
+    class Addons
+      class BrowserStack < Base
+
+        BROWSERSTACK_HOME = '$HOME/.browserstack'
+        BROWSERSTACK_BIN_FILE = 'BrowserStackLocal'
+        BROWSERSTACK_BIN_URL = 'https://www.browserstack.com/browserstack-local'
+        ENV_KEY = 'BROWSERSTACK_KEY'
+        ENV_LOCAL = 'BROWSERSTACK_LOCAL'
+
+        def before_before_script
+          browserstack_key = access_key.to_s
+          sh.fold "browserstack.local.install" do
+            if browserstack_key.empty?
+              sh.echo "You provided an invalid access_key.", ansi: :red
```

Comment:
"access_key is invalid" is more succinct and to the point. (I also think `You`—the person who is reading the logs—is not necessarily the person who provided the key.)